### PR TITLE
BUG: iterparse of read_xml not parsing duplicate element and attribute names

### LINKED ISSUE #47343: BUG: iterparse on read_xml overwrites with attributes on child elements
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
from tempfile import NamedTemporaryFile
import pandas as pd

XML = '''
<issue>
  <type>BUG</type>
  <reporter type="newbie">Emanuel</reporter>
</issue>
'''.encode('utf-8')

with NamedTemporaryFile() as tempfile:
    tempfile.write(XML)
    tempfile.flush()
    df = pd.read_xml(tempfile.name, iterparse={'issue': ['type']})
    issue_type = df.iloc[0]['type']
    print(f"issue_type: expecting BUG, got {issue_type}")
```

### Issue Description

The parsing code parses attributes on all child elements, rather than just on the main element.

This is probably unintentional - in the example above, the user is most likely interested in the type of the incident rather than the type attribute on some child element.

This feature is planned for 1.5.0 and not yet released. See: https://github.com/pandas-dev/pandas/pull/45724#issuecomment-1154200083

### Expected Behavior

Attributes are parsed only on the toplevel element.

### Installed Versions

5465f54f7640cbb4be9b223846823d4935142431

### PR DESCRIPTION
**Pull request by ParfaitG**
- [X] closes #47343 (Replace xxxx with the Github issue number)
- [X] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [X] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [X] Added an entry in the latest `doc/source/whatsnew/v1.5.0.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/io/xml.py [hunk 1]
```diff
@@ -413,11 +413,21 @@ def _iterparse_nodes(self) -> list[dict[str, str | None]]:
                     row = {}
 
             if row is not None:
-                for col in self.iterparse[row_node]:
-                    if curr_elem == col:
-                        row[col] = elem.text.strip() if elem.text else None
-                    if col in elem.attrib:
-                        row[col] = elem.attrib[col]
+                if self.names:
```
Comment (phofl) [inline pandas/io/xml.py:416]: Could you share this?

Comment (ParfaitG) [reply pandas/io/xml.py:340]: Good point. I refactored to use code once in base class of the parsers.

Comment (phofl) [reply pandas/io/xml.py:340]: This is a bit confusing now. Could you split the refactoring and the bug fix into 2 prs? Hard to review like this

Comment (ParfaitG) [reply pandas/io/xml.py:416]: Will revert back to bug fix only and on subsequent PR refactor to remove repetitive code between parsers.

**Review by phofl — APPROVED** [review]

File: doc/source/whatsnew/v1.5.0.rst [hunk 2]
```diff
@@ -874,6 +874,7 @@ I/O
 - Bug in :class:`StataWriter` where value labels were always written with default encoding (:issue:`46750`)
 - Bug in :class:`StataWriterUTF8` where some valid characters were removed from variable names (:issue:`47276`)
 - Bug in :meth:`DataFrame.to_excel` when writing an empty dataframe with :class:`MultiIndex` (:issue:`19543`)
+- Bug in :func:`read_xml` when reading XML with duplicate element and attribute names (:issue:`47343`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.5.0.rst:877]: Just to confirm, this PR fixes a bug that is separate from https://github.com/pandas-dev/pandas/pull/45724 which will come out in 1.5?

If this fixes something related to https://github.com/pandas-dev/pandas/pull/45724, I don't think we need a whatsnew note here

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @ParfaitG

**Pull request merged**